File: backend/src/main/java/com/kushal/ecommerce/backend/service/interfaces/AuthService.java
package com.kushal.ecommerce.backend.service.interfaces;


import com.kushal.ecommerce.backend.dto.request.LoginRequest;
import com.kushal.ecommerce.backend.dto.request.RegisterRequest;
import com.kushal.ecommerce.backend.dto.response.AuthResponse;

public interface AuthService {

    void register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: frontend/src/service/Authservice.ts
import axiosInstance from "../api/axios";
import type{
  AuthResponse,
  LoginRequest,
  RegisterRequest,
} from "../types/auth/types";

export const loginUser = async (
  data: LoginRequest
): Promise<AuthResponse> => {

  const response = await axiosInstance.post(
    "/auth/login",
    data
  );

  return response.data;
};

export const registerUser = async (
  data: RegisterRequest
) => {

  const response = await axiosInstance.post(
    "/auth/register",
    data
  );

  return response.data;
};